### DISCUSSION TITLE
[Future Improvements] simplify new features for normal users

### DISCUSSION THREAD

**Discussion by pheiduck** [wg-easy/wg-easy#1179] [Ideas]
Just a list of things that should be improved for the next version:

- Inital setup when Docker first starts (web UI based)
- Trigger down the complexity of the password hash (should also be done via the web UI and the magic happens in the background) Partially done: #1180

... the list can still grow, just some ideas from the codeowner that we are discussing internally.

Everyone is welcome to make these things happen <3

**Comment by tetuaoro** [comment]
Hi contributors,

I’d like to propose some future improvements for this project based on the recent opened/closed issues :

### 3. User groups

Implement user grouping to manage permissions, such as enabling or disabling internet access for specific groups.

### 5. Firewall interface

Provide a seamless interface for users to manage rules.

###

I’d love to hear your thoughts and any additional suggestions. Let’s work together !

**Reply by offizium-berndstorath** [reply]
###  Send Config from UI

Users should be able to send the Config directly from the UI to the Device/Person to install. Via Email, Discord, Telegram, etc

### Avatar

Be able to configure Avatar using something like https://www.dicebear.com/.
Subavatar to configure the Device Type (Phone, Laptop, etc)

### Alerts

Send Alerts on connect, disconnect over discord, telegram, etc via Email, etc

### Last Handshake

Reload Client after that Time to remove IP from Ram

### Traffic Stats

Store Traffic Stats in Database. To be able to disable Client if Quota Limit is reached. Be able to reset them

### Documentation

- Api Documentation
- Iptable Rules for common use cases (https://github.com/wg-easy/wg-easy/discussions/579)
   
###  Postponed (could be added down the road):
- multiple Users that are able to login and create e.g. upto x peers
- Multiple Instance support (multiple instances to bypass e.g. geo block) (https://github.com/wg-easy/wg-easy/issues/1137)
- Be able to search clients by name, or filter by e.g. connections status
- Tests (E2E, Unit)
- Security (Login with Yubikey, 2FA, (file) permissions, fail2ban)
- OAuth, OpenID, Support for Auth Provider (e.g. Authentik), LDAP
- Re-add Dockerless https://github.com/wg-easy/wg-easy/discussions/1432#discussioncomment-11189854
- DNS (connect the clients wie Client Name, connect to containers via container name)

**Reply by BobWs** [reply]
I highly recommend implementing passkey authentication in WG-Easy using Pocket-ID.

Pocket-ID is a simple and easy-to-use OpenID Connect (OIDC) provider that allows users to authenticate with their passkeys to the services.

**Comment by pheiduck** [comment]
Message from @WeeJeWel:


> @pheiduck could you please replace https://github.com/wg-easy/wg-easy/blob/master/How_to_generate_an_bcrypt_hash.md with a web-based onboarding screen? 
> 
> It would be so much better aligned with the ‘easy’ part of ‘wg-easy’.
> 
> Something like:
> 
> Welcome to WireGuard Easy! Let’s get you set up. 
> 
> Please enter a password for the Admin UI
> […..]
> [Continue] 
> 
> 
> This will then generate and save a bcrypt in node.js and save it to wg0.json. This removed the need for an environment variable entirely.

**Comment by pheiduck** [comment]


**Reply by offizium-berndstorath** [reply]
This is what #1228 is for. But its stuck on #1244 (which could never be accepted)

**Reply by tetuaoro** [reply]
let's be positive :+1:

**Reply by offizium-berndstorath** [reply]
#1228 alone is sufficient to justify #1244

**Reply by pheiduck** [reply]
I am asked @WeeJeWel again due #1244 is required for #1228 

nothing else I can do at the moment.

**Comment by offizium-berndstorath** [comment]


**Reply by offizium-berndstorath** [reply]
I personally also think we should ditch the inline editing and configure the client through a extra page or dialog. can easily be done with #1244

**Reply by tetuaoro** [reply]
> I think the long term goal of `wg-easy` is that it also has more sophisticated/expert features...

I'm agree with that.
 
> With everything through the ui no more configs, env vars or generating password hashes, everything would be configured through the ui. Also stuff like dns and such (for power users) could be configured through the ui but with sane defaults so everything would work even with no config

The *ultime* goal.

**Reply by offizium-berndstorath** [reply]
> The _ultime_ goal.

With features outlined here https://github.com/wg-easy/wg-easy/discussions/1179#discussioncomment-10084528 and here https://github.com/wg-easy/wg-easy/discussions/1179#discussioncomment-10264297, I think its possible and not that far off. Stuff like DNS or Keep Alive should be quite easy, especially with #1244

**Reply by kaaax0815** [reply]
@pheiduck Would like to hear your opinion. (I also sent you an email (maybe old address?, because you don't have any way to reach out to you))

**Comment by JoyZzzzz** [comment]
I would like to additionally support the AmneziaWG protocol. And when creating a user, be able to choose which protocol will be used: Wireguard or AmneziaWG.
Currently, AmnewziaWG has two modifications based on wg-easy.
https://github.com/w0rng/amnezia-wg-easy
https://github.com/spcfox/amnezia-wg-easy

**Reply by kaaax0815** [reply]
As this would heavily influence the Dockerfile we would have to support 2 separate images.
And there is already the problem that people don't submit enough information.
So I can't see us supporting this in the near future

**Comment by austonpramodh** [comment]
Would love some way to disable AUTH on v15 with an env variable since I would use an auth proxy, or maybe support autologin with HTTP Basic headers(I tried already, but it didn't work).

**Comment by kaaax0815** [comment]
Closing this as feature request are done through issues. and only implemented if requested because of time constraints
